Remove the usage of `set_var`

ChenRuiwei opened the pull request:
`std::env::set_var` is marked as unsafe starting in the 2024 Edition.

https://doc.rust-lang.org/edition-guide/rust-2024/newly-unsafe-functions.html

ChenRuiwei committed:
Fix set_var in log-with-tracing.md

`std::env::set_var` is marked as unsafe starting in the 2024 Edition.

https://doc.rust-lang.org/edition-guide/rust-2024/newly-unsafe-functions.html

ChenRuiwei committed:
Fix set_var in async-template-counter

ChenRuiwei committed:
Fix compile error in async-template-counter

ChenRuiwei committed:
Fix indent in log-with-tracing.md

joshka commented on code/templates/async-template-counter/src/utils.rs:
```diff
@@ -98,12 +98,14 @@ pub fn initialize_logging() -> Result<()> {
+        .or_else(|_| std::env::var(LOG_ENV.clone()))
+        .unwrap_or_else(|_| format!("{}=info", env!("CARGO_CRATE_NAME"))),
+    )
+  };
```
It's probably better to remove the set_var call instead of making this unsafe, and configuring EnvFilter below properly.

orhun replied:
I'm okay with either way, we can also add a comment why this is unsafe etc.

Maybe it would be nice to show both ways?

joshka replied:
Not really - it's there because EnvFilter uses RUST_LOG in the call later down, but it doesn't have to. Envfilter has [a way to choose the env variable to use](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.Builder.html#method.with_env_var), and it has [a way to set a default directive](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.Builder.html#method.with_default_directive) that's used if the env variable is not set. And it has [a way to return an error if the variable is not set](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.Builder.html#method.try_from_env). I.e. all the things that this code tries to do here are handled by correctly using `EnvFilter::builder`. The set_var call is unnecessary.

ChenRuiwei replied:
Great insights! Solved in [the latest commit](https://github.com/ratatui/ratatui-website/pull/886/commits/6f75f87e007145498dcf961b0e754fec37d34856).

ChenRuiwei committed:
Refactor initialize_logging by removing set_var

The usage of [`from_default_env`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html#method.from_default_env)
is equivelent to

```rust
use tracing_subscriber::filter::{EnvFilter, LevelFilter};

EnvFilter::builder()
    .with_default_directive(LevelFilter::ERROR.into())
    .from_env_lossy()
```

While in this case the `with_default_directive` will never be reached
because we always set a **vaild** crate level filter string to `RUST_LOG`.

And [`from_env_lossy`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.Builder.html#method.from_env_lossy)
[uses `parse_lossy` internally](https://docs.rs/tracing-subscriber/latest/src/tracing_subscriber/filter/env/builder.rs.html#168-171)

orhun commented on code/templates/async-template-counter/src/utils.rs:
```diff
@@ -98,19 +98,16 @@ pub fn initialize_logging() -> Result<()> {
-      .or_else(|_| std::env::var(LOG_ENV.clone()))
-      .unwrap_or_else(|_| format!("{}=info", env!("CARGO_CRATE_NAME"))),
-  );
+  let filter_string = std::env::var("RUST_LOG")
```
nit:
```suggestion
  let log_filter = std::env::var("RUST_LOG")
```

ChenRuiwei committed:
Rename filter_string to log_filter
